publish-mqtt/src/homie.rs

```diff
@@ -303,6 +344,55 @@ impl HomieDevice {
     }
 }
 
+/// Legacy stats extension.
+struct HomieStats {
+    requests_tx: Sender<Request>,
+    device_base: String,
+    start_time: Instant,
+    last_sent: Instant,
+}
+
+impl HomieStats {
+    fn new(requests_tx: Sender<Request>, device_base: String) -> Self {
+        let now = Instant::now();
+        Self {
+            requests_tx,
+            device_base,
+            start_time: now,
+            last_sent: now - STATS_INTERVAL,
+        }
+    }
+
+    /// Send initial topics.
+    async fn start(
+        requests_tx: &Sender<Request>,
+        device_base: &str,
+    ) -> Result<(), SendError<Request>> {
+        publish_retained(
+            requests_tx,
+            format!("{}/$stats/interval", device_base),
+            STATS_INTERVAL.as_secs().to_string(),
+        )
+        .await
+    }
+
+    /// Send latest stats, if enough time has passed since last time.
+    async fn refresh(&mut self) -> Result<(), SendError<Request>> {
+        let now = Instant::now();
+        if now > self.last_sent + STATS_INTERVAL {
```

Comment:
Ideally, I would ask for the initial last_sent to be set to the minimum value that Instant can take, but Rust doesn't seem to expose this. Failing that, let's at least make sure that things work if time stands still (it's common to freeze time in tests in python. I don't know how common this is in rust).
```suggestion
        if now >= self.last_sent + STATS_INTERVAL {
```